### ISSUE TITLE
[RFC] Add feature labels for GitHub issues

### ISSUE THREAD

**Issue by DanailH** [mui-org/material-ui-x#3596]
## Why

If I remember correctly, sometime in Q4 we discussed adding additional `DataGrid` feature-specific labels in GitHub (suggested by @oliviertassinari) so we can organize the issues better. This will also help us track which features are used more than others, which once are more stable and which aren't so much.
In addition, it will also provide a visual aid for the code owners to focus more on what they are familiar with.

The Core follows a similar organizational tactic. There they have component-specific labels, so implementing feature-specific labels will increase the repo familiarity when people open issues on `material-ui-x` repo.

## Proposition

I propose we start with matching what features we have listed in the docs and then add more or remove some if needed. Below are what we can start with.
- `feature: Layout`
- `feature: Editing`
- `feature: Sorting`
- `feature: Filtering`
- `feature: Pagination`
- `feature: Selection`
- `feature: Events`
- `feature: Export`
- `feature: Components`
- `feature: Localization`
- `feature: Scrolling`
- `feature: Virtualization`
- `feature: Accessibility`

The tricky part comes when we need to handle the `Rows` and `Columns` sections. There are two approaches where we for now just add `feature: Rows` and `feature: Columns` labels or we list what is inside those pages. For example, for the `Columns` section, we will have: `feature: Column types`, `feature: Column selector`, `feature: Column reorder` etc. Of course, we can apply that same logic to the other pages but it may be too much to start with.

## Next steps

I will add the labels but I would like to agree on the list before I do so. So if we go with this proposition a potential issue related to the pagination will have the following labels as an example:
`bug` `component: DataGrid` `feature: Pagination`

**Comment by oliviertassinari** [comment]
> feature: Accessibility

We already have the "accessibility" label, it sounds too generic.

> feature: Localization

We already have the "l10n" label, it sounds too generic. Also, it's l10n and i18n are two **different** things. The former is about the content, then the latter about the infrastructure to make it happen (AFAIK).

For the groups, I think that they should be aligned with what we use in https://www.notion.so/mui-org/fb47797b32fc42cfb28b4205f77d6265?v=f572ec29fffa45649be345adf4212210, to see with the rest of the team what works best.

**Comment by DanailH** [comment]
For the groups/components I would propose to also rename the current `component: data grid` to `component: DataGrid` to be consistent with the others.

Regarding the `accessibility` and `i10n` we can rename them to `feature: Accessibility` and `feature: i10n` for consistency but keeping them as they are is also fine as they will match those in the core repo.

**Comment by oliviertassinari** [comment]
> For the groups/components I would propose to also rename the current component: data grid to component: DataGrid to be consistent with the others.

OK, I had this in my TODO for a long time. It's a good opportunity for me to continue the conversation that I started at https://github.com/mui-org/material-ui/pull/29930. @m4theushw Has been asking for it for some time 👼. I didn't merge this PR in MUI Core because I wanted to align it with MUI X too.

### Problem

So, let's start with the problem statement:

1. In https://github.com/mui-org/material-ui-x/pull/3606, I add the standard badge header we have on all the components. It has a link to access all the feedback from the community:

  <img width="606" alt="Screenshot 2022-01-13 at 12 29 43" src="https://user-images.githubusercontent.com/3165635/149322447-3b0ff354-643e-48a8-b314-82305c01ca9f.png">

  But what should "Feedback" link to? I would argue that it should link to all the data grid issues, the plan is not relevant. However, currently ["component: DataGrid"](https://github.com/mui-org/material-ui-x/labels/component%3A%20data%20grid) is not applied when the discussion is about a pro or premium plan features.
  
 2. Because we have no label that encompasses all the data grid discussion when developers have to search for a specific problem, they have to search for **each plan** `"component: DataGrid"`, `"component: DataGridPro"`, `"component: DataGridPremium"`. I would argue that this is not great.
 
### Proposed solution
 
Regarding the solution. I would propose the same one as in https://github.com/mui-org/material-ui/pull/29930. We have one generic data grid label that we apply to differentiate the issues between data grid, pickers (and the future components). We had specific labels for JS/TS modules that are relevant to us. We add the `pro plan` and `premium plan` labels to signal the plan.

I can illustrate how it will look like with https://github.com/mui-org/material-ui-x/pull/3604. It would have the following labels:

  - component: data grid
  - pro plan
  - feature: Tree Data

**Comment by DanailH** [comment]
Ok, the proposed solution does make sense. I agree that we need a label to differentiate the data grid issues from the date picker once and at the same time be able to easily filter all the data grid-related issues. The only tweak I would propose is the following:
 - for `component: data grid` we can have just this one, so remove the `component: DataGridPro`. We can change the one we already have to `component: DataGrid` just for consistency but it's also ok to keep it as is. Also create label for `component: DatePicker` or `component: date picker`
 - instaed of `pro plan` do `plan: Community`, `plan: Pro`, `plan: Premium`
 - for the feature, I think we are all in agreement
 - we keep the `i10n` and `accessibility` as is, we will reuse those labels for the `DatePicker`.

So to summarise the same ticket  #3604 will have the following labels: 
- `component: DataGrid` or `component: data grid`
- `plan: Pro`
- `feature: Tree Data`

**Comment by oliviertassinari** [comment]
@DanailH Sounds great

**Comment by flaviendelangle** [comment]
As a follow up on yesterday's discussion
Can we remove the `question` label put by default in the `pro-support` template ?
For me it's a misusage of the label.

**Comment by oliviertassinari** [comment]
> Can we remove the question label put by default in the pro-support template?

What's the purpose of this template?

From my perspective, "Pro plan: support ❔" is meant for pro users that need a feedback on something that is neither a bug (should use "Bug report 🐛") nor a features request (should use "Feature request 💄"). We have introduced it so we could reduce the need to use the x@mui.com Zendesk support channel.

**Comment by flaviendelangle** [comment]
My point was more that `question` = The team asked a question to the user, not that the issue is about a question to the team (that is "support").
At least that's how I see it used.

For the interest of the template itself, for me it can make sense.

**Comment by m4theushw** [comment]
To give my 5 cents about the feature labels. We can start with only the following labels: 

- `feature: Editing`
- `feature: Sorting`
- `feature: Filtering`
- `feature: Pagination`
- `feature: Selection`
- `feature: Fluid width`
- `feature: Other`

These ones I'm sure that they will be useful since we keep receiving bug reports related to them. Bug reports that don't fail into any category, I would use `feature: Other`. Once we see a new category emerging we go back and re-tag the "other" issues.

---

I would keep the question template because the number of issues opened with it is low. We can still handle them. Also, we can have a sense of the pains that the users are facing when using the documentation. About using the `question` label to signal that the team asked a question, I don't know if this is correct. During triage, if something is missing I add `status: incomplete` and leave it there until all questions have been answered. I'm not used to change the labels after asking something.

**Comment by oliviertassinari** [comment]
> My point was more that question = The team asked a question to the user, not that the issue is about a question to the team (that is "support").
At least that's how I see it used.

@flaviendelangle ahhh, yes, agree, this label description is confusing. We have the same confusing description in MUI Core. I believe that the "question" label is coming from the core. Ther, it was designed to label StackOverflow like questions from developers, but ones that we still wanted to answer in GitHub. For the questions that we don't want to answer, we use "support".

Regarding feedback that goes the other way around, maintainers asking the community, we have used "discussion" and "waiting for upvotes", "need further information" labels.

Here, what I woudo suggest is that we update the question label description to be clearer.

**Comment by DanailH** [comment]
To be fair we can keep the current `question` label as is but as @m4theushw suggested use the `status: incomplete` when we are asking the user a question. 
Then again the chare from `question` to `support: Question` also makes sense.

**Comment by DanailH** [comment]
Alright, I've added the new labels updated old ones, and adjusted some of the existing labels descriptions. I'm closing this issue. In the future, we would need to revise what has been marked as `feature: Other` and create additional labels if needed.

In addition, we also need to make one more label -> `label: date picker` but I think this one can wait a bit (it will also need feature labels for it but I think it's too early for that as well).

**Comment by flaviendelangle** [comment]
For the date picker related labels, I can create them just before I start migrating issues from the core.

**Comment by oliviertassinari** [comment]
I had a look at all the open issues, I have tried to apply the labels, where relevant.

**Comment by oliviertassinari** [comment]
A data point that hasn't been raised so far. AG Grid shares the feature labels they use. It could maybe be of some inspiration in the future:

<img width="335" alt="Screenshot 2022-02-11 at 11 19 35" src="https://user-images.githubusercontent.com/3165635/153574864-75c7dcd9-1c3b-4503-a768-45c91bc9192f.png">

https://www.ag-grid.com/pipeline/
